pkg/disasterrecovery/disasterrecovery.go

```diff
@@ -0,0 +1,163 @@
+		APIVersion: "builtin",
+		Kind:       "LabelTransformer",
+		Metadata: OverlySimpleMetadata{
+			Name: "backup-label-transformer",
+		},
+		Labels: labels,
+		FieldSpecs: []kustomizetypes.FieldSpec{
+			{
+				Path:               "metadata/labels",
+				CreateIfNotPresent: true,
+			},
+			{
+				Path: "spec/template/metadata/labels",
+				Gvk: resid.Gvk{
+					Version: "v1",
+					Kind:    "ReplicationController",
+				},
+				CreateIfNotPresent: true,
+			},
+			{
+				Path: "spec/template/metadata/labels",
+				Gvk: resid.Gvk{
+					Kind: "Deployment",
+				},
+				CreateIfNotPresent: true,
+			},
+			{
+				Path: "spec/template/metadata/labels",
+				Gvk: resid.Gvk{
+					Kind: "ReplicaSet",
+				},
+				CreateIfNotPresent: true,
+			},
+			{
+				Path: "spec/template/metadata/labels",
+				Gvk: resid.Gvk{
+					Kind: "DaemonSet",
+				},
+				CreateIfNotPresent: true,
+			},
+			{
+				Path: "spec/template/metadata/labels",
+				Gvk: resid.Gvk{
+					Group: "apps",
+					Kind:  "StatefulSet",
+				},
+				CreateIfNotPresent: true,
+			},
+			{
+				Path: "spec/template/metadata/labels",
+				Gvk: resid.Gvk{
+					Group: "batch",
+					Kind:  "Job",
+				},
+				CreateIfNotPresent: true,
+			},
+			{
+				Path: "spec/jobTemplate/metadata/labels",
+				Gvk: resid.Gvk{
+					Group: "batch",
+					Kind:  "CronJob",
+				},
+				CreateIfNotPresent: true,
+			},
+			{
+				Path: "spec/jobTemplate/spec/template/metadata/labels",
+				Gvk: resid.Gvk{
+					Group: "batch",
+					Kind:  "CronJob",
+				},
+				CreateIfNotPresent: true,
+			},
+		},
+	}
+
+	b, err := yaml.Marshal(labelTransformer)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal disaster recovery label transformer")
+	}
+
+	return b, nil
+}
+
+func EnsureLabelTransformer(archiveDir string, additionalLabels map[string]string) error {
+	labelTransformerExists := false
+
+	k, err := k8sutil.ReadKustomizationFromFile(filepath.Join(archiveDir, "overlays", "midstream", "kustomization.yaml"))
+	if err != nil {
+		return errors.Wrap(err, "failed to read kustomization file from midstream")
+	}
+
+	for _, transformer := range k.Transformers {
+		if transformer == LabelTransformerFileName {
+			labelTransformerExists = true
+			break
+		}
+	}
+
+	if !labelTransformerExists {
+		drLabelTransformerYAML, err := GetLabelTransformerYAML(additionalLabels)
+		if err != nil {
+			return errors.Wrap(err, "failed to get disaster recovery label transformer yaml")
+		}
+
+		absFilename := filepath.Join(archiveDir, "overlays", "midstream", LabelTransformerFileName)
+
+		if err := ioutil.WriteFile(absFilename, drLabelTransformerYAML, 0644); err != nil {
+			return errors.Wrap(err, "failed to write disaster recovery label transformer yaml file")
+		}
+
+		k.Transformers = append(k.Transformers, LabelTransformerFileName)
+
+		if err := k8sutil.WriteKustomizationToFile(*k, filepath.Join(archiveDir, "overlays", "midstream", "kustomization.yaml")); err != nil {
```

Comment:
This belongs in the midstream package.  Is it possible to move it there?

Reply:
@divolgin I assume you're referring to the whole function, not just this step, right?